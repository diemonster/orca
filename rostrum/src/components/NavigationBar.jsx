import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { login, logout, setUsername } from '../actions/authActions';
import './navigation-bar.css';


export class NavigationBar extends React.Component {
  constructor(props) {
    super(props);

    this.handleLogin = this.handleLogin.bind(this);
    this.handleLogout = this.handleLogout.bind(this);
  }

  handleLogin() {
    const { dispatchLogin } = this.props;
    dispatchLogin();
  }

  handleLogout() {
    const { dispatchLogout } = this.props;
    dispatchLogout();
  }

  render() {
    const { dispatchSetUsername, isAuthenticated, username } = this.props;

    if (isAuthenticated) {
      return (
        <div className="navigation-bar">
          <ul>
            <li className="profile">
              {username}
            </li>
            <li className="auth">
              <button type="button" onClick={this.handleLogout}>
              Log Out
              </button>
            </li>
          </ul>
        </div>
      );
    }

    if (username) {
      dispatchSetUsername('');
    }

    return (
      <div className="navigation-bar">
        <ul>
          <li className="auth">
            <button type="button" onClick={this.handleLogin}>
            Log In
            </button>
          </li>
        </ul>
      </div>
    );
  }
}

NavigationBar.propTypes = {
  dispatchLogin: PropTypes.func.isRequired,
  dispatchLogout: PropTypes.func.isRequired,
  dispatchSetUsername: PropTypes.func.isRequired,
  isAuthenticated: PropTypes.bool.isRequired,
  username: PropTypes.string.isRequired,
};

const mapStateToProps = state => ({
  isAuthenticated: state.auth.isAuthenticated,
  username: state.auth.username,
});

const mapDispatchToProps = dispatch => ({
  dispatchLogin: () => dispatch(login()),
  dispatchLogout: () => dispatch(logout()),
  dispatchSetUsername: username => dispatch(setUsername(username)),
});

export default connect(mapStateToProps, mapDispatchToProps)(NavigationBar);
