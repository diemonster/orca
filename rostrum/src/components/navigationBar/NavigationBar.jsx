import React from 'react';
import './navigation-bar.css';
import PropTypes from 'prop-types';

function NavigationBar(props) {
  const { login, logout, name } = props;
  if (logout) {
    return (
      <div className="navigation-bar">
        <ul>
          <li className="home"><a href="/">Home</a></li>
          <li className="profile">
            <button type="button" onClick={logout}>
              {name ? `${name}, Log Out` : 'Log Out'}
            </button>
          </li>
        </ul>
      </div>
    );
  }

  return (
    <div className="navigation-bar">
      <ul>
        <li className="profile">
          <button type="button" onClick={login}>
            Log In
          </button>
        </li>
      </ul>
    </div>
  );
}

NavigationBar.propTypes = {
  login: PropTypes.func,
  logout: PropTypes.func,
  name: PropTypes.string,
};

NavigationBar.defaultProps = {
  login: null,
  logout: null,
  name: null,
};

export default NavigationBar;
