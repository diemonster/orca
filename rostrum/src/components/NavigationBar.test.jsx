import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { NavigationBar } from './NavigationBar';

Enzyme.configure({ adapter: new Adapter() });

describe('components', () => {
  let props;

  beforeEach(() => {
    props = {
      dispatchLogin: jest.fn(),
      dispatchLogout: jest.fn(),
      isAuthenticated: false,
      username: '',
    };
  });

  describe('NavigationBar', () => {
    it('should render the login view when not authenticated', () => {
      const navigationBar = shallow(<NavigationBar {...props} />);
      expect(navigationBar).toMatchSnapshot();
    });

    it('should render the logout view when authenticated', () => {
      props.isAuthenticated = true;
      props.username = 'username';

      const navigationBar = shallow(<NavigationBar {...props} />);
      expect(navigationBar).toMatchSnapshot();
    });

    describe('methods', () => {
      it('handleLogin should dispatch a login action', () => {
        const navigationBar = shallow(<NavigationBar {...props} />);
        navigationBar.find('button').simulate('click');
        expect(props.dispatchLogin).toHaveBeenCalledWith();
      });

      it('handleLogout should dispatch a logout action', () => {
        props.isAuthenticated = true;
        props.username = 'username';

        const navigationBar = shallow(<NavigationBar {...props} />);
        navigationBar.find('button').simulate('click');
        expect(props.dispatchLogout).toHaveBeenCalledWith();
      });
    });
  });
});
