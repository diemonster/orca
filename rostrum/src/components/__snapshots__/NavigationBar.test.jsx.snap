// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components NavigationBar should render the login view when not authenticated 1`] = `
<div
  className="navigation-bar"
>
  <ul>
    <li
      className="auth"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Log In
      </button>
    </li>
  </ul>
</div>
`;

exports[`components NavigationBar should render the logout view when authenticated 1`] = `
<div
  className="navigation-bar"
>
  <ul>
    <li
      className="profile"
    >
      username
    </li>
    <li
      className="auth"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        Log Out
      </button>
    </li>
  </ul>
</div>
`;
